src/cli.ts

```diff
@@ -157,23 +157,32 @@ function runWithPaths(paths: Array<string>, options: CLIOptions, callback: ((err
 
   function processDirectory(path: string): void {
     readdir(path, (err, children) => {
-      if (err) { errors.push(err); }
-      else {
-        pending.unshift(
-          ...children
-            .filter(child => {
-              if (options.modernizeJS) {
-                return child.endsWith('.js');
-              } else {
-                return child.endsWith('.coffee') ||
-                  child.endsWith('.litcoffee') ||
-                  child.endsWith('.coffee.md');
-              }
-            })
-            .map(child => join(path, child))
-        );
+      if (err) {
+        errors.push(err);
+        return;
       }
-      processNext();
+      children.forEach((child, index) => {
+        stat(join(path, child), (err, info) => {
+          if (err) {
+            errors.push(err);
+            return;
+          }
+
+          if (
+            info.isDirectory() ||
+            (options.modernizeJS && child.endsWith('.js')) ||
+            // Extensions .coffee || .litcoffee || .coffee.md
+            /\.(lit)?coffee(\.md)?$/.test(child)
+          ) {
+            pending.push(join(path, child));
+          }
+
+          // On last run, process next
+          if (children.length === index + 1) {
```

Comment:
Hmm I think this actually has a race condition. All of the stat calls are happening in parallel and might come back in any order, so I think the very last one (which might be non-coffeescript) could come back first, call `processNext` before the others get added, and none of the other files will get processed.